wandb/sdk/launch/utils.py

```diff
@@ -690,7 +690,7 @@ def docker_image_exists(docker_image: str, should_raise: bool = False) -> bool:
 
 def pull_docker_image(docker_image: str) -> None:
     """Pull the requested docker image."""
-    if docker_image_exists(docker_image):
+    if not docker_image.endswith(":latest") and docker_image_exists(docker_image):
```

Comment:
I think this function should do the simplest thing which should just be pull the image, I recommend moving the `latest` special casing into `docker_image_exists` where we consider `latest` to never exist.

Reply:
or actually, maybe we keep the `docker_image_exists` pure and handle this purely in `local_contaienr` runner. We'd need to swallow `LaunchError` for `:latest` in the case where the image already exists

Reply:
That function is actually only used here, but I think it is cleaner if we handle it in local container and have the function do precisely what the name says with minimal logic.